test/Waives.Pipelines.Tests/WorkPoolFacts.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Waives.Pipelines.Tests
+{
+    public class WorkPoolFacts
+    {
+        [Fact]
+        public async Task Post_runs_supplied_func()
+        {
+            var sut = new WorkPool(3);
+            var hasRun = false;
+
+            sut.Post(() =>
+            {
+                hasRun = true;
+                return Task.CompletedTask;
+            });
+            await sut.WaitAsync();
+
+            Assert.True(hasRun);
+        }
+
+        [Theory]
+        [InlineData(1,3)]
+        [InlineData(3,3)]
+        [InlineData(100,100)]
```

Comment:
📝 Spaces needed between the arguments. 

💪 I think it would be good to make this _obviously_ a performance test:
```suggestion
        [InlineData(1000, 1000)]
```